collection-scripts/gather_monitoring

```diff
@@ -1,29 +1,208 @@
 #!/bin/bash
 
+# Gather Monitoring Metrics and Dashboards.
+# - Discovery and save Grafana dashboards, extracting the metrics' datapoints from Prometheus.
+# - Gather custom metrics (when defined on Config variables).
+# - Discovery metrics from prefix name.
+#
+# Config variables:
+# - GATHER_MONIT_START_DATE: starting human readable date string (date -d). Default: "7 days ago".
+# - GATHER_MONIT_END_DATE: ending human readable date string (date -d). Default: "now".
+# - GATHER_MONIT_QUERY_STEP: metric resolution to get from Prometheus. Default: "1m".
+# - GATHER_MONIT_CUSTOM_METRICS: list of additional metrics name to be collected. Default: Undefined
+# - GATHER_MONIT_DISCOVERY_PREFIXES: discovery metrics by prefixes. Default: ''
+# - GATHER_MONIT_TOKEN: optional service account token to gather Prometheus and Grafana API. Default: Undefined
+#
+# To create the ConfigMap:
+# $ echo -e 'GATHER_MONIT_START_DATE="15 days ago"\nGATHER_MONIT_QUERY_STEP="5m"' > env
+# $ echo -e 'GATHER_MONIT_CUSTOM_METRICS="up"' > env
+# $ echo -e 'GATHER_MONIT_DISCOVERY_PREFIXES="apiserver_"' > env
+# $ oc create configmap must-gather-env -n openshift-monitoring --from-file=env=env
```

Comment:
How does this configmap get injected into the must-gather pod?

Reply:
User will need to remember to clean this up as well don't they? 

It may be less magic if user can point to any configmap rather than create on in `openshift-monitoring` e.g. 

`oc adm must-gather -- /usr/bin/gather-monitoring-data  --use-cm=ns/cm-name`

Reply:
> How does this configmap get injected into the must-gather pod?

Upon created, the script load it [here : `function get_env`](https://github.com/openshift/must-gather/pull/214/files#diff-a34d5a93515e38dfc675ecd82be2d0c92c9a384475fc1a6b157b311a262daebeR52-R64)

> User will need to remember to clean this up as well don't they?

Unfortunately if you want to customize the "metrics collector", the CM need to be created before the execution and manually cleaned after it finished.

I didn't think in a life cycle of this CM. Could be a good idea send it as argument, but it still needing to be manually created/removed. Initially I thought in remove this CM after execution (on script) but I not sure if it will break the principle of MG to be a "super-get tool".

Reply:
As you suggested, I renamed the script from `gather_monitoring` to `gather-monitoring-data`. I am implementing the flags.
https://github.com/mtulio/must-gather/pull/21/commits/62c19c3f9acd85a655446d0594f8f72103d01002